### ISSUE TITLE
Usage with Oak, custom routes and middleware

### ISSUE THREAD

**Issue by Octo8080X** [exhibitionist-digital/ultra#43]
Thank you for the wonderful tool.

I have been using it since 0.6.
In 0.6, router had been exported.
After 0.7, will router be export or API extend not be support?

Source code at my app.
https://github.com/Octo8080/Twitter-Timeline-app

**Comment by mashaal** [comment]
Hey @Octo8080X - Yes, from v0.7.0 onwards, `oak` will no longer be a dependency of `ultra`. We are aiming to reduce the amount of dependencies moving forward, while making it easier to support differing targets.

One of the ways we are looking to support `oak` is with a generic request handler, which will be  similar to this pseudo-code. I think there are more benefits by having full management of `oak` outside of `ultra`.

```js
import { Application, Router } from "https://deno.land/x/oak/mod.ts";
import { ultraHandler} from "https://deno.land/x/ultra/mod.ts";
import importmap from "./importmap.json" assert { type: "json" };

const app = new Application();
const router = new Router();

router
  .get("/custom-route", (context) => {
    context.response.body = "#1";
  })
  .get("/*", (context) => {
    context.response = await ultraHandler({importmap, request: context.request})
  });

await app.listen({ port: 8000 });
```

I will add this to the v0.8.0 milestone, as this request handler is not exposed yet.

**Comment by Octo8080X** [comment]
Thank you.
I am very happy.
We are looking forward to the release of the generic request handler in 0.8.0.

**Comment by mashaal** [comment]
Merged and will be released in v0.8.0
